File: backend-go/internal/service/chapter_service.go
package service

import (
	"github.com/ruankao/gaojia-backend-go/internal/dto"
	"github.com/ruankao/gaojia-backend-go/internal/repository"
)

type ChapterService struct {
	chapterRepo *repository.ChapterRepository
}

func NewChapterService(chapterRepo *repository.ChapterRepository) *ChapterService {
	return &ChapterService{chapterRepo: chapterRepo}
}

func (s *ChapterService) Tree() ([]dto.ChapterTreeNode, error) {
	return s.chapterRepo.FindTree()
}

File: backend/src/main/java/com/ruankao/gaojia/service/ChapterService.java
package com.ruankao.gaojia.service;

import com.ruankao.gaojia.dto.ChapterTreeNode;
import com.ruankao.gaojia.repository.ChapterRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ChapterService {

    private final ChapterRepository chapterRepository;

    public ChapterService(ChapterRepository chapterRepository) {
        this.chapterRepository = chapterRepository;
    }

    public List<ChapterTreeNode> tree() {
        return chapterRepository.findTree();
    }
}
